### PR TITLE
Create api/projects/search

### PR DESCRIPTION
**Pull request by julienlancelot**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/project/ws/SearchAction.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    this.support = support;
+  }
+
+  @Override
+  public void define(WebService.NewController context) {
+    WebService.NewAction action = context.createAction(ACTION_SEARCH)
+      .setSince("6.3")
+      .setDescription("Search for projects or views.<br>" +
+        "Requires 'System Administrator' permission")
+      .setInternal(true)
```
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/project/ws/SearchAction.java]: why internal?

File: server/sonar-server/src/main/java/org/sonar/server/project/ws/SearchAction.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+      .addPagingParams(100)
```
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/project/ws/SearchAction.java]: no check on pagination

File: server/sonar-server/src/main/java/org/sonar/server/project/ws/SearchAction.java (hunk 1 above)
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/project/ws/SearchAction.java]: Why not ?

File: server/sonar-server/src/main/java/org/sonar/server/project/ws/SearchAction.java (hunk 2 above)
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/project/ws/SearchAction.java]: where ?

**Pull request merged**